### ISSUE TITLE
[mini-css-extract-plugin] Conflicting order between:

### ISSUE THREAD

**Issue by tigermax139** [ant-design/ant-design#15696]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
Can't reproduce

### Steps to reproduce
During build process react-create-app
I receive warnings regarding `Conflicts order between` styles in antd lib

### What is expected?
-

### What is actually happening?
```
chunk 3 [mini-css-extract-plugin]
Conflicting order between:
 * css ./node_modules/css-loader??ref--7-oneOf-3-1!./node_modules/postcss-loader/src??postcss!./node_modules/antd/lib/empty/style/index.css
 * css ./node_modules/css-loader??ref--7-oneOf-3-1!./node_modules/postcss-loader/src??postcss!./node_modules/antd/lib/select/style/index.css
```

| Environment | Info |
|---|---|
| antd | 3.15.2 |
| React | ^16.8.2 |
| System | Ubuntu 16.04 |
| Browser | Not related to browser |

---
Could you please fix/sort your styles

**Comment by afc163** [comment]
Duplicate of #14895

**Comment by tigermax139** [comment]
Sorry, but you are not right!
[#14895](https://github.com/ant-design/ant-design/issues/14895) 
has problem with others your modules. Button and drop down.

**Comment by CGreenburg** [comment]
We're experiencing this error as well. The only solution was to override the CI environment variable and set it to false in my CI Pipeline in order to get the build to pass.

**Comment by tigermax139** [comment]
So, I investigate this issue more deeply.  And sure that this is problem is MY styles. AFAIK, `ant` generate one bundle css or chunk per route  (depends on webpack configuration). And it's not same as in `Material-UI` which inject and encapsulate styles per component. In case with `ant` you need to provide this encapsulation by self. It's can be unique class-wrapper for each component or just define all your styles in sigle file

**Comment by illyaMs** [comment]
For those who uses `react-app-rewired` or analogues and sticks to go `disable mini-css-extract-plugin warning` way:

1. make sure you have `"mini-css-extract-plugin": "^0.8.0"`, otherwise add it (there's an extra flag in this version to skip ordering warnings)
2. add this ~(awful hack)~ to your `config-overrides.js`:
```
  // force react-scripts to use newer version of `mini-css-extract-plugin` and ignore css ordering warnings
  // (related to issue: https://github.com/facok/create-react-app/issues/5372)
  for (let i = 0; i < config.plugins.length; i++) {
    const p = config.plugins[i]
    if(!!p.constructor && p.constructor.name === MiniCssExtractPlugin.name) {
      const miniCssExtractOptions = {...p.options, ignoreOrder: true}
      config.plugins[i] = new MiniCssExtractPlugin(miniCssExtractOptions)
      break
    }
  }
```

**Comment by bulutfatih** [comment]
Is there any certain way to fix this?
I'm using Next.js + Ant Design and I see a lot of warnings like this.

> chunk styles [mini-css-extract-plugin]
> Conflicting order between:
> css ./node_modules/@zeit/next-css/node_modules/css-loader??ref--5-

Also, pages sometimes don't display properly. I think it is because of this conflict error.

**Comment by ChuckJonas** [comment]
+ 1 on using with Next.js

**Comment by jonahfang** [comment]
+1 on using with Next.js!

**Comment by kilnamkim** [comment]
+1 on using with Next.js!

**Comment by loq24** [comment]
+1 on using Next.js! 

@ChuckJonas @jonahfang @kilnamkim did you find any solution to this issue? Thanks!

**Comment by ChuckJonas** [comment]
@loq24 Nothing yet... ☹️

**Comment by loq24** [comment]
Ok thanks for the update @ChuckJonas

**Comment by saltz** [comment]
@loq24 @ChuckJonas Just wondering if you guys found a solution or workaround?

**Comment by loq24** [comment]
@saltz I had to hide the warning messages for now until there's a fix. I'm using the `webpack-filter-warnings-plugin` package.

**Comment by devuxer** [comment]
Also seeing this with gatsbyjs

**Comment by kamilml** [comment]
+1 on using with Next.js!

any solution? :(

![](https://i.imgur.com/oGZID5T.png)

**Comment by saltz** [comment]
@kamilml I resolved my issues after upgrading to the latest version of antd (~4.2.5) and nextjs (~9.4.2). Bear in mind that this upgrade could potentially break other integrations you currently have and will require you to update your `next.config.js`.

**Comment by kamilml** [comment]
> @kamilml I resolved my issues after upgrading to the latest version of antd (~4.2.5) and nextjs (~9.4.2). Bear in mind that this upgrade could potentially break other integrations you currently have and will require you to update your `next.config.js`.

@saltz  I have this, still with the same problem :(

```json
    "dependencies": {
        "@next/bundle-analyzer": "^9.4.4",
        "@reduxjs/toolkit": "^1.3.6",
        "@zeit/next-less": "^1.0.1",
        "antd": "^4.3.1",
        "antd-img-crop": "^3.3.0",
        "isomorphic-unfetch": "^3.0.0",
        "next": "^9.4.1",
        "next-antd-aza-less": "^1.0.2",
        "react": "^16.13.1",
        "react-dom": "^16.13.1",
        "react-redux": "^7.2.0"
    },
    "devDependencies": {
        "null-loader": "^4.0.0"
    }
```

**Comment by 510846** [comment]
> I think it is because of this

do you found the solution?

**Comment by Mehran91z** [comment]
Same here, +1 on using Nextjs + Emotionjs and Ant Design `less configuration`...
Is there any update there?

**Comment by oliverwu** [comment]
Nextjs + Ant Design
I solved this paroblem by replace `import { Button } from 'antd'` with `import Button from 'antd/lib/button'`

**Comment by qiwang97** [comment]
> My environment is Nextjs + Ant Design
> I solved this problem by replace `import { Button } from 'antd'` with `import Button from 'antd/lib/button'`

fix button but others come. Too hard to deal with antd and nextjs especially when upgrade any of those.

**Comment by devuxer** [comment]
@afc163, Can you reopen?

**Comment by fillipo** [comment]
+1

**Comment by GermaVinsmoke** [comment]
If related to Antd then do these 2 things - 

- If you import styles by specifying the `style` option of babel-plugin-import, change it from `'css'` to `true`, which will import the `less` version of antd.

- If you import styles from `'antd/dist/antd.css'`, change it to `antd/dist/antd.less`.

The `next.config.js` part can be checked from this [nextjs-css-less-antd](https://github.com/GermaVinsmoke/nextjs-css-less-antd)

**Comment by devuxer** [comment]
@GermaVinsmoke ,

Already have babel-plugin-import set to `style: true`. Still have this problem. (Note: I'm using Gatsby, not next, but I believe the issue occurs regardless.)

**Comment by THPubs** [comment]
I think this issue needs to be opened. It's still happening.

**Comment by qiwang97** [comment]
The simple way is just adding [webpack-filter-warnings-plugin](https://www.npmjs.com/package/webpack-filter-warnings-plugin) to ignore this warning if no other effect.

```js
    config.plugins.push(
      new FilterWarningsPlugin({
        exclude: /Conflicting order/,
      })
    )
```

**Comment by THPubs** [comment]
@qiwang97 I don't think that's a good idea. I'm missing some styles during the first page load. They load only after re-loading.

**Comment by THPubs** [comment]
https://github.com/ant-design/ant-design/issues/14895#issuecomment-489380963 If this comment is true, then we can safely ignore the warnings I think 🤔

**Comment by hassansaadfr** [comment]
I found a solution for this problem ( antd + less + next )

**next.config.js**

```js
const withLess = require('@zeit/next-less')

module.exports = withLess({
  lessLoaderOptions: {
    javascriptEnabled: true
  },
  webpack: (config, { isServer }) => {
    if (isServer) {
      const antStyles = /antd\/.*?\/style.*?/
      const origExternals = [...config.externals]
      config.externals = [
        (context, request, callback) => {
          if (request.match(antStyles)) return callback()
          if (typeof origExternals[0] === 'function') {
            origExternals[0](context, request, callback)
          } else {
            callback()
          }
        },
        ...(typeof origExternals[0] === 'function' ? [] : origExternals)
      ]

      config.module.rules.unshift({
        test: antStyles,
        use: 'null-loader'
      })
    }
    return config
  }
})
```

**.babelrc**

```
{
  "presets": ["next/babel"]
}
```

**pages/_app.js**

```js
import '../styles/antd-custom.less' // includes antd style and our customization
import React from 'react'
import { Button } from 'antd'

function MyApp (props) {

  return (
    <Button>OK</Button>
  )
}

export default MyApp
```

**antd-custom.less**

```
@import "~antd/dist/antd.less";
@primary-color: #52c41a;
```

**Comment by conioX** [comment]
ane news about this issue?

**Comment by ashbuilds** [comment]
Facing same issue.

**Comment by baj84** [comment]
Same issue

**Comment by hassansaadfr** [comment]
@conioX @ashbuilds @baj84 Try my solution

**Comment by siman** [comment]
> @conioX @ashbuilds @baj84 Try my solution 

Does your solution support CSS modules + LESS?

**Comment by hassansaadfr** [comment]
> > @conioX @ashbuilds @baj84 Try my solution
> 
> Does your solution support CSS modules + LESS?

With nextjs ?

**Comment by hassansaadfr** [comment]
> > @conioX @ashbuilds @baj84 Try my solution
> 
> Does your solution support CSS modules + LESS?

if your webpack is correctly configured thats should work

**Comment by siman** [comment]
@hassansaadfr that is the issue, that we failed to get it work all together:
Ant + Less + Next + CSS modules.

**Comment by hassansaadfr** [comment]
> @hassansaadfr that is the issue, that we failed to get it work all together:
> Ant + Less + Next + CSS modules.

```js
const withLess = require('@zeit/next-less')
const withPlugins = require('next-compose-plugins')

const path = require('path')

const plugins = [
  [withLess({
    lessLoaderOptions: {
      javascriptEnabled: true,
    },
    webpack: (config, { isServer }) => {
      if (isServer) {
        const antStyles = /antd\/.*?\/style.*?/
        const origExternals = [...config.externals]
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback()
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback)
            } else {
              callback()
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ]

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        })
      }

      const builtInLoader = config.module.rules.find((rule) => {
        if (rule.oneOf) {
          return (
            rule.oneOf.find((deepRule) => {
              return deepRule.test && deepRule.test.toString().includes('/a^/');

            }) !== undefined
          );
        }
        return false;
      });

      if (typeof builtInLoader !== 'undefined') {
        config.module.rules.push({
          oneOf: [
            ...builtInLoader.oneOf.filter((rule) => {
              return (rule.test && rule.test.toString().includes('/a^/')) !== true;
            }),
          ],
        });
      }

      config.resolve.alias['@'] = path.resolve(__dirname);
      return config;
    }
  })]
]

const nextConfig = {
  env: {
    name: "blabla"
  }
}

module.exports = withPlugins(plugins, nextConfig)

```

This configuration work 👍

**Comment by ashbuilds** [comment]
@hassansaadfr Hi, Just tried the last fix you posted. Still the same warning.

**Comment by hassansaadfr** [comment]
@ashbuilds Check my example on codesandbox [here](https://codesandbox.io/s/nextjs-antd-less-css-modules-9qbei?file=/pages/index.js)
i have just one warning during the build but this is normal 
![image](https://user-images.githubusercontent.com/37713877/94815027-5bb18500-03fa-11eb-952f-aa8d42ab98c1.png)

**Comment by qiwang97** [comment]
Nextjs v10 supports[  importing CSS from Third Party React Components](https://nextjs.org/blog/next-10#importing-css-from-third-party-react-components).
Any update to solve this issue here?

**Comment by Mohammed-Poolwla** [comment]
Hi @hassansaadfr 
The conflict between **antd** .less files has been solved but not in my projects .less files...

**Comment by hassansaadfr** [comment]
@Mohammed-Poolwla show me your code. Or send me a codesandbox 👍

**Comment by siman** [comment]
+1 to show the working code

**Comment by hassansaadfr** [comment]
Hi @siman @Mohammed-Poolwla check my codesandbox for the working code [codesanbox](https://codesandbox.io/s/nextjs-antd-withcss-withless-k46wm)

**Comment by Mohammed-Poolwla** [comment]
Hi @hassansaadfr 
please check these warnings

![image](https://user-images.githubusercontent.com/22235509/97968352-89686000-1de4-11eb-9a49-e88e270cf163.png)

**Comment by hassansaadfr** [comment]
@Mohammed-Poolwla Can i see your .babelrc file ?

**Comment by Mohammed-Poolwla** [comment]
@hassansaadfr here is the code

{
    "presets": ["next/babel"]
}

**Comment by hassansaadfr** [comment]
@Mohammed-Poolwla  Can i see your next.config.js and your antd-custom.less too ?

**Comment by Mohammed-Poolwla** [comment]
> @Mohammed-Poolwla Can i see your next.config.js and your antd-custom.less too ?

Can we connect via skype or any medium? so that I show you my code.

**Comment by Mohammed-Poolwla** [comment]
Here are both files 

**next.config.js**

```

/* eslint-disable */
const withLess = require("@zeit/next-less");
const withCss = require("@zeit/next-css");
module.exports = withCss({
  cssModules:true,
  images: {
    domains: ['qa-be.profitley.com'],
  },
  ...withLess({
    lessLoaderOptions: {
      javascriptEnabled: true
    },
    webpack: (config, { isServer }) => {
      if (isServer) {
        const antStyles = /antd\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === "function") {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === "function" ? [] : origExternals)
        ];
        config.module.rules.unshift({
          test: antStyles,
          use: "null-loader"
        });
      }
      return config;
    }
  })
});

```
**antd-custom.less**

```
@import "~antd/dist/antd.less";
@import "./utils/utils.less";
// @import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

@font-size-base: 12px;

@font-family: 'Roboto', sans-serif;
@code-family: 'Roboto', sans-serif;

@primary-color: #0048BA;
@secondary-color: #FF8000;
// @layout-body-background: #52c41a;
@layout-header-height: 40px;
@border-radius-base: 2px;

@btn-default-color: #fff;
@btn-default-bg: @secondary-color;
@btn-default-border: @secondary-color;

@radio-button-bg: #FFF;

```

**Comment by Zonassoft** [comment]
[nextjs 9.4.4, antd 4.7.0, styled-components, react-intl, next-fonts]

I had problems in components styles at first build and I needed to do page refresh almost always. I tested many variants and it seems this configuration fixed the issue for good. I don't really know if the trick is in the "importLoaders: 3" variable.
PS: I'm still have the conflicting styles warnings !
I'm putting the entire configuration here with other modules for in case anyone uses them too.

**next.config.js**

```
/* eslint-disable */
const AntdDayjsWebpackPlugin = require("antd-dayjs-webpack-plugin");
const withLess = require("@zeit/next-less");
const lessToJS = require("less-vars-to-js");
const withCSS = require("@zeit/next-css");
const withFonts = require("next-fonts");
const fs = require("fs");
const path = require("path");
const nextEnv = require("next-env");
const dotenvLoad = require("dotenv-load");

// .env support
dotenvLoad();
const withNextEnv = nextEnv();

// Where your antd-custom.less file lives
const themeVariables = lessToJS(
  fs.readFileSync(path.resolve(__dirname, "./assets/antd-custom.less"), "utf8")
);

const withTM = require("next-transpile-modules")([
  "@formatjs/intl-relativetimeformat",
  "@formatjs/intl-utils",
  "react-intl",
  "intl-format-cache",
  "intl-messageformat-parser",
  "intl-messageformat",
]);

module.exports = withNextEnv(
  withTM(
    withFonts(
      withCSS({
        cssModules: true,
        ...withLess({
          lessLoaderOptions: {
            // sourceMap: true,
            javascriptEnabled: true,
            modifyVars: themeVariables, // make your antd custom effective
            importLoaders: 3,
          },
          cssLoaderOptions: {
            importLoaders: 3,
            localIdentName: "[local]___[hash:base64:5]",
          },
          webpack: (config, { isServer }) => {
            config.plugins = config.plugins || [];
            if (isServer) {
              const antStyles = /antd\/.*?\/style.*?/;
              const origExternals = [...config.externals];
              config.externals = [
                (context, request, callback) => {
                  if (request.match(antStyles)) return callback();
                  if (typeof origExternals[0] === "function") {
                    origExternals[0](context, request, callback);
                  } else {
                    callback();
                  }
                },
                ...(typeof origExternals[0] === "function"
                  ? []
                  : origExternals),
              ];

              config.module.rules.unshift({
                test: antStyles,
                use: "null-loader",
              });
            }

            config.plugins.push(new AntdDayjsWebpackPlugin());
            return config;
          },
        }),
      })
    )
  )
);

```
**.babelrc**
```
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "import",
      {
        "libraryName": "antd",
        "style": true
      }
    ],
    [
      "styled-components",
      {
        "ssr": true,
        "displayName": true,
        "preprocess": true
      }
    ]
  ]
}

```
**next-env.d.ts**
```
/// <reference types="next" />
/// <reference types="next/types/global" />
```

**Comment by jlozanoher** [comment]
[nextjs 9.4.4, antd 4.7.0, styled-components, react-intl, next-fonts]

I had problems in components styles at first build and I needed to do page refresh almost always. I tested many variants and it seems this configuration fixed the issue for good. I don't really know if the trick is in the "importLoaders: 3" variable.
PS: I'm still have the conflicting styles warnings !
I'm putting the entire configuration here with other modules for in case anyone uses them too.

**next.config.js**

```
/* eslint-disable */
const AntdDayjsWebpackPlugin = require("antd-dayjs-webpack-plugin");
const withLess = require("@zeit/next-less");
const lessToJS = require("less-vars-to-js");
const withCSS = require("@zeit/next-css");
const withFonts = require("next-fonts");
const fs = require("fs");
const path = require("path");
const nextEnv = require("next-env");
const dotenvLoad = require("dotenv-load");

// .env support
dotenvLoad();
const withNextEnv = nextEnv();

// Where your antd-custom.less file lives
const themeVariables = lessToJS(
  fs.readFileSync(path.resolve(__dirname, "./assets/antd-custom.less"), "utf8")
);

const withTM = require("next-transpile-modules")([
  "@formatjs/intl-relativetimeformat",
  "@formatjs/intl-utils",
  "react-intl",
  "intl-format-cache",
  "intl-messageformat-parser",
  "intl-messageformat",
]);

module.exports = withNextEnv(
  withTM(
    withFonts(
      withCSS({
        cssModules: true,
        ...withLess({
          lessLoaderOptions: {
            // sourceMap: true,
            javascriptEnabled: true,
            modifyVars: themeVariables, // make your antd custom effective
            importLoaders: 3,
          },
          cssLoaderOptions: {
            importLoaders: 3,
            localIdentName: "[local]___[hash:base64:5]",
          },
          webpack: (config, { isServer }) => {
            config.plugins = config.plugins || [];
            if (isServer) {
              const antStyles = /antd\/.*?\/style.*?/;
              const origExternals = [...config.externals];
              config.externals = [
                (context, request, callback) => {
                  if (request.match(antStyles)) return callback();
                  if (typeof origExternals[0] === "function") {
                    origExternals[0](context, request, callback);
                  } else {
                    callback();
                  }
                },
                ...(typeof origExternals[0] === "function"
                  ? []
                  : origExternals),
              ];

              config.module.rules.unshift({
                test: antStyles,
                use: "null-loader",
              });
            }

            config.plugins.push(new AntdDayjsWebpackPlugin());
            return config;
          },
        }),
      })
    )
  )
);

```
**.babelrc**
```
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "import",
      {
        "libraryName": "antd",
        "style": true
      }
    ],
    [
      "styled-components",
      {
        "ssr": true,
        "displayName": true,
        "preprocess": true
      }
    ]
  ]
}

```
**next-env.d.ts**
```
/// <reference types="next" />
/// <reference types="next/types/global" />
```

**Comment by hassansaadfr** [comment]
@jlozanoher I think you have the warning because you import the antdesign style in .babelrc and in next.config.js.

Try to remove import of Less in .babelrc and in next.config.js.

In your _app.js import your custom less file, and in your less file import default antdesign less.


next.config.js :

```js
/* eslint-disable */
const AntdDayjsWebpackPlugin = require("antd-dayjs-webpack-plugin");
const withLess = require("@zeit/next-less");
const lessToJS = require("less-vars-to-js");
const withCSS = require("@zeit/next-css");
const withFonts = require("next-fonts");
const fs = require("fs");
const path = require("path");
const nextEnv = require("next-env");
const dotenvLoad = require("dotenv-load");

// .env support
dotenvLoad();
const withNextEnv = nextEnv();

// Where your antd-custom.less file lives
//const themeVariables = lessToJS(
//  fs.readFileSync(path.resolve(__dirname, "./assets/antd-custom.less"), "utf8")
//);

const withTM = require("next-transpile-modules")([
  "@formatjs/intl-relativetimeformat",
  "@formatjs/intl-utils",
  "react-intl",
  "intl-format-cache",
  "intl-messageformat-parser",
  "intl-messageformat",
]);

module.exports = withNextEnv(
  withTM(
    withFonts(
      withCSS({
        cssModules: true,
        ...withLess({
          lessLoaderOptions: {
            // sourceMap: true,
            javascriptEnabled: true,
           // modifyVars: themeVariables, // make your antd custom effective
            importLoaders: 3,
          },
          cssLoaderOptions: {
            importLoaders: 3,
            localIdentName: "[local]___[hash:base64:5]",
          },
          webpack: (config, { isServer }) => {
            config.plugins = config.plugins || [];
            if (isServer) {
              const antStyles = /antd\/.*?\/style.*?/;
              const origExternals = [...config.externals];
              config.externals = [
                (context, request, callback) => {
                  if (request.match(antStyles)) return callback();
                  if (typeof origExternals[0] === "function") {
                    origExternals[0](context, request, callback);
                  } else {
                    callback();
                  }
                },
                ...(typeof origExternals[0] === "function"
                  ? []
                  : origExternals),
              ];

              config.module.rules.unshift({
                test: antStyles,
                use: "null-loader",
              });
            }

            config.plugins.push(new AntdDayjsWebpackPlugin());
            return config;
          },
        }),
      })
    )
  )
);
```

.babelrc
```
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "styled-components",
      {
        "ssr": true,
        "displayName": true,
        "preprocess": true
      }
    ]
  ]
}
```

your custom less file :
```
@import "~antd/dist/antd.less";
```

And in pages/_app.js import your less file

**Comment by steveswork** [comment]
There is now an NPM package named  **[@webkrafters/ordercss](https://www.npmjs.com/package/@webkrafters/ordercss)** which tackles this issue.

Full disclosure: I initially created it to solve this very problem in one of my apps but decided to expand it and share it with everyone.

If this package helps anyone, please share it with others.

Thanks and good luck!

**Comment by zhi114** [comment]
@hassansaadfr thanks for your solution.But your config will import all ant design css once for all.If I want to import  js and css modularly,how can I setup next.js?

**Comment by kristapsPelna** [comment]
@hassansaadfr +1 on needing this to also work with treeshaking the css.
For treeshaking to work I have this plugin on my .babelrc
`"plugins": [["import", { "libraryName": "antd", "style": true }]]`
But doing this I keep getting the css order warnings

**Comment by hassansaadfr** [comment]
This ils because you load the CSS two times. One time in your less file and one time in babelrc. You should remove the import in babelrc. Check m'y example 👍

**Comment by kristapsPelna** [comment]
@hassansaadfr I mean when I have the import plugin and dont import the ~antd/dist/antd.less on _app.js I do get treeshaking for the CSS. I get only the css from the antd components that I use. BUT, I get the CSS order warnings.

If I use the approach you mentioned, then I dont get the warnings, BUT the whole antd css is imported and its not being treeshaken.

**Comment by alienzhangyw** [comment]
@kristapsPelna Exactly same issue here!

**Comment by rashadrivera** [comment]
+1 on Reopening

I'm having the same issue in my VueJS projects.  Below are some of my config details

**package.json**
{
	"name": "TEST",
	"version": "1.2.3",
	"private": true,
	"description": "Test 001",
	"author": {
		"name": "Abbott n Castello"
	},
	"scripts": {
		"serve": "vue-cli-service serve --copy",
		"build": "vue-cli-service build --mode production",
		"build_dev": "vue-cli-service build --mode development",
		"lint": "vue-cli-service lint --no-fix",
		"lint_fix": "vue-cli-service lint",
		"clean": "",
		"test": "jest",
		"test_watch": "jest --watchAll"
	},
	"dependencies": {
		"axios": "0.21.1",
		"core-js": "3.6.5",
		"moment": "2.29.0",
		"name-of": "1.0.2",
		"node": "14.13.0",
		"sass": "1.26.11",
		"sass-loader": "10.0.2",
		"vue": "2.6.12",
		"vue-class-component": "7.2.6",
		"vue-property-decorator": "9.0.0",
		"vue-router": "3.4.5",
		"vue-the-mask": "0.11.1",
		"vue-unique-id": "3.1.1",
		"vuetify": "2.3.12",
		"vuetify-confirm": "2.0.6",
		"vuetify-loader": "1.6.0",
		"vuex": "3.5.1",
		"vuex-class": "0.3.2",
		"vuex-router-sync": "5.0.0"
	},
	"devDependencies": {
		"@fortawesome/fontawesome-free": "5.15.0",
		"@types/jest": "26.0.14",
		"@types/vue-the-mask": "0.11.0",
		"@typescript-eslint/eslint-plugin": "4.3.0",
		"@typescript-eslint/parser": "4.3.0",
		"@vue/cli-plugin-babel": "4.5.6",
		"@vue/cli-plugin-eslint": "4.5.6",
		"@vue/cli-plugin-router": "4.5.6",
		"@vue/cli-plugin-typescript": "4.5.6",
		"@vue/cli-service": "4.5.6",
		"@vue/eslint-config-airbnb": "5.1.0",
		"@vue/eslint-config-typescript": "5.1.0",
		"@vue/test-utils": "1.1.0",
		"babel-core": "7.0.0-bridge.0",
		"babel-jest": "^26.6.3",
		"eslint": "7.10.0",
		"eslint-plugin-vue": "7.0.0",
		"jest": "26.5.3",
		"jest-transform-stub": "^2.0.0",
		"ts-jest": "26.4.1",
		"typescript": "4.0.3",
		"vue-cli-plugin-vuetify": "2.0.7",
		"vue-jest": "3.0.7",
		"vue-template-compiler": "2.6.12",
		"vuetify-loader": "^1.7.1"
	},
	"postcss": {
		"plugins": {
			"autoprefixer": {}
		}
	}
}

**vue.config**
/* eslint-disable */
// const VuetifyLoaderPlugin = require('vuetify-loader/lib/plugin');

module.exports = {

    publicPath: process.env.BASE_URL,
    css: {
        loaderOptions: {
            scss: {
                // this has no affect until we use transpileDependencies: ["vuetify"]
                additionalData: '@import "~@/scss/style.scss";',
            },
        },
    },
    // FROM:https://cli.vuejs.org/config/#devserver-proxy
    devServer: {

        // https://webpack.js.org/configuration/dev-server/
        compress: true,
        // transpileDependencies: ['vuetify'],
    },
    chainWebpack: (config) => {
        /// /config.plugin('html')
        /// /	.tap(args => {
        /// /		//args[0].WEB_SITE_ROOT = process.env.VUE_APP_WEB_SITE_ROOT;
        /// /		args[0].title = 'Loading';
        /// /		return args;
        /// /	});
        // config.plugin('VuetifyLoaderPlugin').tap(args => [{
        //	match(originalTag, { kebabTag, camelTag, path, component }) {
        //		if (kebabTag.startsWith('core-')) {
        //			return [camelTag, `import ${camelTag} from '@/components/core/${camelTag.substring(4)}.vue'`]
        //		}
        //	}
        // }]);

        // Deletes build cache so that ESLint rules will apply everytime we build
        // Known issue in Vue-CLI.  Solution gained from issue posting
        // SEE: https://github.com/vuejs/vue-cli/issues/5399#issuecomment-625794727
        config.module.rule('vue').uses.delete('cache-loader');
        config.module.rule('js').uses.delete('cache-loader');
        config.module.rule('tsx').uses.delete('cache-loader');
        config.module.rule('ts').uses.delete('cache-loader');
        //config.module.rule('eslint').use('eslint-loader').options({
        //  fix: true,
        //});
    },
    parallel: true,
};

**babel.config**
/* eslint-disable */
module.exports = {
  presets: [
    ['@vue/cli-plugin-babel/preset', {
      useBuiltIns: 'entry',
      debug: false, // <= EMITS DEBUG POLYFILL MESSAGES
      //   targets: {
      //     node: 'current',
      /// /	browsers: [
      /// /		"IE >= 11"
      /// /	]
      // },
    }],
  ],
};

**Comment by kaioduarte** [comment]
@afc163 @zombieJ can you reopen this? This is still an issue :/

**Comment by ionpetro** [comment]
I am still having this issue and I can not seem to find any solution yet.

Using next, ant and less.

**Comment by sunkid1995** [comment]
> Is there any certain way to fix this?
> I'm using Next.js + Ant Design and I see a lot of warnings like this.
> 
> > Conflicting order between:
> > 
> > * css ./node_modules/@zeit/next-css/node_modules/css-loader??ref--5-2!./node_modules/less-loader/dist/cjs.js??ref--5-3!./node_modules/antd/lib/typography/style/index.less
> > * css ./node_modules/@zeit/next-css/node_modules/css-loader??ref--5-2!./node_modules/less-loader/dist/cjs.js??ref--5-3!./node_modules/antd/lib/pagination/style/index.less
> > * css ./node_modules/@zeit/next-css/node_modules/css-loader??ref--5-2!./node_modules/less-loader/dist/cjs.js??ref--5-3!./node_modules/antd/lib/spin/style/index.less
> 
> Also, pages sometimes don't display properly. I think it is because of this conflict error.

+1 on using with Next.js. I have same issue!

**Comment by mateuszoracz** [comment]
After 5h i found and solved this problem by:
Update from "antd": "^4.12.2" to "antd": "^4.16.6"

and change import:

from: import { Card } from 'antd';
to: import Card from 'antd/lib/card';

Stack: react-app-rewired without next.js ofc

The worst part is that the warning says absolutely nothing about which component has the problem. In my case only Card, but we'll see.

**Comment by ddanngg** [comment]
With NextJS 11.0, I've found excellent success with the following plugin
[https://github.com/SolidZORO/next-plugin-antd-less](https://github.com/SolidZORO/next-plugin-antd-less)

Repo example: [https://github.com/SolidZORO/mkn](https://github.com/SolidZORO/mkn)

**Comment by ionpetro** [comment]
Same

**Comment by xiaoxixi6633** [comment]
vue 项目 可以这样配置
    
config.plugin('extract-css').tap(args => {
      args[0].ignoreOrder = true
      return args
    })
